main.go

```diff
@@ -0,0 +1,292 @@
+			kingpin.Fatalf(err.Error())
+		}
+		p.Opts = cli.Options{
+			Names:    *namesOutput,
+			Long:     *longOutput,
+			Area:     *area,
+			Service:  *service,
+			Resource: *resource,
+		}
+		cli.RunGetTask(p, *outputFmt)
+	case projectSetCmd.FullCommand():
+		p, err := cli.FindProject(*projectSetID, *projectSetDomain)
+		if err != nil {
+			kingpin.Fatalf(err.Error())
+		}
+		cli.RunSetTask(p, projectSetQuotas)
+	case projectSyncCmd.FullCommand():
+		p, err := cli.FindProject(*projectSyncID, *projectSyncDomain)
+		if err != nil {
+			kingpin.Fatalf(err.Error())
+		}
+		cli.RunSyncTask(p)
+	}
+}
+
+// quotas contains the aggregate parsed quota values.
+type quotas cli.Quotas
+
+// Set implements the kingpin.Value interface.
+func (q *quotas) Set(value string) error {
+	value = strings.TrimSpace(value)
+	// tmp contains the different components of a single parsed quota value. This makes it easier to refer
+	// to individual components and pass them to the cli.Quotas map
+	var tmp cli.Resource
+
+	// separate the quota value from its identifier
+	idfVal := strings.SplitN(value, "=", 2)
+	if len(idfVal) != 2 {
+		return fmt.Errorf("expected a quota in the format: service/resource=value(unit), got '%s'", value)
+	}
+
+	// separate service and resource
+	srvRes := strings.SplitN(idfVal[0], "/", 2)
+	if len(srvRes) != 2 {
+		return fmt.Errorf("expected service/resource, got '%s'", idfVal[0])
+	}
+	srv := srvRes[0]
+	tmp.Name = srvRes[1]
+
+	// separate quota value and comment (if one was given)
+	valCom := strings.SplitN(idfVal[1], ":", 2)
+	if len(valCom) > 1 {
+		if valCom[1] != "" {
+			tmp.Comment = strings.TrimSpace(valCom[1])
+		}
+	}
+
+	// separate quota's value from its unit (if one was given)
+	rx := regexp.MustCompile(`^([0-9]+)([A-Za-z]+)?$`)
+	match := rx.MatchString(valCom[0])
+	if !match {
+		return fmt.Errorf("expected a quota value with optional unit in the format: 123Unit, got '%s'", valCom[0])
+	}
+
+	// rxMatchedList: []string{"entire regex matched string", "quota value", "unit (empty, if no unit given)"}
+	rxMatchedList := rx.FindStringSubmatch(valCom[0])
+	intVal, err := strconv.ParseInt(rxMatchedList[1], 10, 64)
+	if err != nil {
+		return fmt.Errorf("could not parse quota value: '%s'", rxMatchedList[1])
+	}
+	tmp.Value = intVal
+
+	tmp.Unit = limes.UnitNone
+	if rxMatchedList[2] != "" {
+		switch rxMatchedList[2] {
+		case "B":
+			tmp.Unit = limes.UnitBytes
+		case "KiB":
+			tmp.Unit = limes.UnitKibibytes
+		case "MiB":
+			tmp.Unit = limes.UnitMebibytes
+		case "GiB":
+			tmp.Unit = limes.UnitGibibytes
+		case "TiB":
+			tmp.Unit = limes.UnitTebibytes
+		case "PiB":
+			tmp.Unit = limes.UnitPebibytes
+		case "EiB":
+			tmp.Unit = limes.UnitExbibytes
+		default:
+			return fmt.Errorf("acceptable units: ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB'], got '%s'", rxMatchedList[2])
+		}
+	}
+
+	if _, exists := (*q)[srv]; !exists {
+		(*q)[srv] = make([]cli.Resource, 0)
+	}
+	(*q)[srv] = append((*q)[srv], tmp)
```

Comment:
You don't need the `if !exists` part. `append()` will happily accept `nil` as its first argument and, in that case, allocate a completely fresh slice.